lib/eslint.js

```diff
@@ -657,10 +666,16 @@ module.exports = (function() {
         // process initial config to make it safe to extend
         config = prepareConfig(config || {});
 
-        ast = parse(text.replace(/^#!([^\r\n]+)/, function(match, captured) {
-            shebang = captured;
-            return "//" + captured;
-        }), config);
+        if (textOrParseResult.ast) {
+            // if a pregenerated ast was supplied, use it
+            ast = textOrParseResult.ast;
```

Comment:
Agreed - I stuck with the original naming, but strictly speaking it should be ast+tokens+other stuff. That seems to cause a fair bit of confusion, but my code is correct - the usage of ast in the input is intentionally wrong for consistency.